src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -43,6 +47,13 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     request.Path.ToUriComponent(),
                     request.QueryString.ToUriComponent());
 
+                // use a specific SSL port if it is provided in the options,
+                // clear it otherwise to use an empty default.
+                var uri = new UriBuilder(newUrl);
```

Comment:
Suspect the only case where the original `newUrl` can be used is if both `SslPort` is `null` and there's no port in the `request.Host` string. Do we have a way to inspect `request.Host` that's less icky than instantiating a `UriBuilder`?